storey/utils.py

```diff
@@ -344,3 +345,25 @@ def find_filters(partitions_time_attributes, start, end, filters, filter_column)
         filters,
         filter_column,
     )
+
+
+def get_combined_filters(datetime_filters=None, filters=None):
+    datetime_filters = copy.deepcopy(datetime_filters) or [[]]
+    filters = copy.deepcopy(filters) or []
+    or_filter_exists = False
+    filter_tuple_exists = False
+
+    for or_filter in datetime_filters:
+        for tuple_filter in filters:
+            if tuple_filter:
+                filter_tuple_exists = True
+                if or_filter:
+                    or_filter_exists = True
+                    or_filter.append(tuple_filter)
+            elif or_filter:
+                or_filter_exists = True
+
+    #  datatime filters doesn't exists or [[],[]] :
+    if not or_filter_exists and filter_tuple_exists:
+        return [filters]
+    return datetime_filters
```

Comment:
Can't all of this just be replaced with
```python
    if datetime_filters and filters:
        datetime_filters[0] += filters
    return datetime_filters
```
?

Reply:
nope, if datetime_filters are not exist, and filter exists, we will need to return filters.
moreover, we want to add `additional_filters` to each list (=or_filter).

Reply:
But there is only ever one outer filter that we create for `datetime_filters`.

Reply:
look at `test_filter_before_after_partitioned_random`

Reply:
and also there are situations were there are not datetime_filters.

Reply:
Why? It passes with the 3 line implementation I posted above.

Reply:
fixed.